hack/jenkins/benchmark-dockerized.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+retry() {
+  for i in {1..5}; do
+    "$@" && return 0 || sleep $i
+  done
+  "$@"
+}
+
+# Runs benchmark integration tests, producing JUnit-style XML test
+# reports in ${WORKSPACE}/artifacts. This script is intended to be run from
+# kubekins-test container with a kubernetes repo mapped in. See
+# k8s.io/test-infra/scenarios/kubernetes_verify.py
+
+export PATH=${GOPATH}/bin:${PWD}/third_party/etcd:/usr/local/go/bin:${PATH}
+
+retry go get github.com/tools/godep && godep version
+retry go get github.com/jstemmer/go-junit-report
+
+# Disable the Go race detector.
+export KUBE_RACE=" "
+# Disable coverage report
+export KUBE_COVER="n"
+# Produce a JUnit-style XML test report for Jenkins.
+export KUBE_JUNIT_REPORT_DIR=${WORKSPACE}/artifacts
+export ARTIFACTS_DIR=${WORKSPACE}/artifacts
+
+cd /go/src/k8s.io/kubernetes
+
+./hack/install-etcd.sh
+
+make test-integration \
+    WHAT="$*" \
+    KUBE_TEST_ARGS="-- -test.run='XXX' -benchtime=5s -bench=. -benchmem"
```

Comment:
Will that work for scheduler benchmark?

Reply:
I had to slightly change the command (mainly remove the benchtime as it was creating some issues). Checked it locally and seems to works:

```
$ make test-integration WHAT=./test/integration/scheduler_perf KUBE_TEST_ARGS="-run='XXX' -bench=. -benchmem"
...
BenchmarkScheduling/100Nodes/0Pods-12 	     500	   2208003 ns/op	  431617 B/op	    4827 allocs/op
...
BenchmarkScheduling/100Nodes/1000Pods-12         	     500	   2092055 ns/op	  458291 B/op	    5345 allocs/op
...
```